### ISSUE TITLE
Ingredients not respecting custom model data/other NBT data.

### ISSUE THREAD

**Issue by ophelovie** [greatericontop/GreatCrafts#4]
**Version info**
GreatCrafts version (/greatcrafts) (make sure you are using the latest!): v1.4.0
Minecraft server software and version (copy the full output of /ver): Pufferfish 1.21.1-18-HEAD@63ce67d

**Describe the bug**
Ingredients are not respecting custom model data/other NBT data.

**To Reproduce**
1. Put an apple with custom model data as an ingredient in a recipe.
2. Save and apply recipe.
3. Attempt to use normal apple to craft the recipe. It will allow you to use it.

**Expected behavior**
Ingredients should respect custom model data/other NBT data.

**Screenshots**
![image_2024-12-02_144827412](https://github.com/user-attachments/assets/2e74a92d-9b27-46a1-8ccb-a388132e75f8)
![image_2024-12-02_144850509](https://github.com/user-attachments/assets/5272ffd2-892b-43d4-afb4-62c00140cbea)
![image_2024-12-02_144908936](https://github.com/user-attachments/assets/95edb4b4-5528-4fc8-9db3-b486756a8039)

**Comment by greatericontop** [comment]
Did you have exact choice on the apple?

**Comment by ophelovie** [comment]
Ahh, how would I go about doing that? I didn't see it on the Modrinth page! Thank you for replying so fast :)

**Comment by ophelovie** [comment]
Oh god never mind I just figured that out. I'm super sorry!
I do see the option for Exact Choice. Is it possible for a way to just check for custom model data and not other factors such as display name? Just incase players rename the item

**Comment by greatericontop** [comment]
That doesn't exist unfortunately, as exact choice is a vanilla feature and it requires a match of all the NBT.

You can try putting the custom item (without a name) in the grid with exact choice and see if a renamed item works though. I haven't tried it but it may or may not work.

**Comment by ophelovie** [comment]
Yeah, I've given that a go and that sadly did not work.
Is it something you would be interested in adding to the plugin?

**Comment by greatericontop** [comment]
It's a bit difficult to do as it requires the plugin to manually check everything but I could try adding something like a "semi-exact choice" once I'm a bit less busy. I think it would be better to just ask your players to not rename the custom items

**Comment by ophelovie** [comment]
I see! no worries at all. Thank you very much for answering my questions - I really appreciate it! And I apologise for submitting this as a bug report to start with!
The functionality i'm looking for isn't there so i will look for something else because I need it for an update this week. It's no worries and I thank you a lot for your time!

**Comment by greatericontop** [comment]
Yeah no problem! If you need it fast my best suggestion is to just not have them rename the item.
